targets/TARGET_TI/TARGET_MSP432/serial_api.c

```diff
@@ -0,0 +1,399 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2019 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* Low-level implementation of uart functionality for MSP432.
+ * This implementation does not (yet) support SERIAL_ASYNCH and SERIAL_FC.
+ */
+
+#if DEVICE_SERIAL
+
+#include "serial_api.h"
+#include "mbed_assert.h"
+#include "mbed_error.h"
+#include <string.h>
+
+/* With SERIAL_ASYNCH, our type serial_s is embedded
+ * into a bigger structure (see serial_api.h). So we
+ * need a macro to extract the serial_s object.
+ */
+#if DEVICE_SERIAL_ASYNCH
+#define SERIAL_S(obj) (&((obj)->serial))
+#else
+#define SERIAL_S(obj) (obj)
+#endif
+
+int stdio_uart_inited = 0; // used in mbed_retarget.cpp
+serial_t stdio_uart;
+
+uart_irq_handler uart_handler = 0;
+uint32_t serial_irq_ids[4] = {0};
+
+/** Initialize the serial peripheral. It sets the default parameters for serial
+ *  peripheral, and configures its specifieds pins.
+ *
+ * @param obj The serial object
+ * @param tx  The TX pin name
+ * @param rx  The RX pin name
+ */
+void serial_init(serial_t *obj, PinName tx, PinName rx) {
```

Comment:
can you fix the coding style in HAL implementation (there are few breakages to the style in this file, some functions look fine). You can use astyle config file and run it through _api.c files

Reply:
Done in latest commit!